cpp/v2/test/api_test_v2.cc

```diff
@@ -0,0 +1,262 @@
+  EXPECT_EQ(ray::Task(add).Remote(ray::Put(1), 3).Get(), 3);
+  EXPECT_EQ(ray::Task(add).Remote(2, ray::Put(1)).Get(), 2);
+  EXPECT_EQ(ray::Task(add).Remote(ray::Put(1), ray::Put(2)).Get(), 0);
+}
+
+TEST(RayApiTestV2, NamespaceNormalTask) {
+  auto obj1 = ray::Task(bar).Remote(2);
+  EXPECT_EQ(obj1.Get(), 2);
+
+  auto obj2 = ray::Task(my_namespace::bar).Remote(2);
+  EXPECT_EQ(obj2.Get(), 3);
+
+  EXPECT_EQ(ray::Task(bar).Remote(2).Get(), 2);
+  EXPECT_EQ(ray::Task(my_namespace::bar).Remote(2).Get(), 3);
+}
+
+TEST(RayApiTestV2, OverlaodNormalTask) {
+  auto obj = ray::Task(RayFunc(overload_func)).Remote();
+  EXPECT_EQ(obj.Get(), 0);
+
+  auto obj1 = ray::Task(RayFunc(overload_func, int)).Remote(2);
+  EXPECT_EQ(obj1.Get(), 2);
+}
+
+TEST(RayApiTestV2, CreateActor) {
+  auto actor_handle = ray::Actor<Base>().Remote();
+  EXPECT_EQ(typeid(actor_handle), typeid(ray::ActorHandle<Base>));
+
+  auto actor_handle2 = ray::Actor<Base>().Remote(1);
+  EXPECT_EQ(typeid(actor_handle2), typeid(ray::ActorHandle<Base>));
+
+  auto actor_handle3 = ray::Actor<Base>().Remote(1, ObjectRef<int>{});
+  EXPECT_EQ(typeid(actor_handle3), typeid(ray::ActorHandle<Base>));
+
+  auto actor_handle4 = ray::Actor(Base::Create).Remote(1);
+  EXPECT_EQ(typeid(actor_handle4), typeid(ray::ActorHandle<Base>));
+}
+
+TEST(RayApiTestV2, ActorTask) {
+  auto obj = ray::Actor<Base>().Remote().Task(&Base::bar).Remote(1);
+  EXPECT_EQ(obj.Get(), 1);
+
+  // Pass ObjectRef<int> to bar
+  auto obj2 = ray::Actor<Base>().Remote().Task(&Base::bar).Remote(ObjectRef<int>{2});
+  EXPECT_EQ(obj2.Get(), 2);
+
+  // Overload function
+  auto obj3 = ray::Actor<Base>()
+                  .Remote()
+                  .Task(RayMemberFunc(&Base::overload_func, cv_none, int))
+                  .Remote(ObjectRef<int>{1});
+  EXPECT_EQ(obj3.Get(), 1);
+
+  auto obj4 = ray::Actor<Base>()
+                  .Remote()
+                  .Task(RayMemberFunc(&Base::overload_func, cv_none, int, int))
+                  .Remote(1, ObjectRef<int>{2});
+
+  EXPECT_EQ(obj4.Get(), 3);
+
+  auto obj5 = ray::Actor<Base>()
+                  .Remote()
+                  .Task(RayMemberFunc(&Base::overload_func, cv_none, int, int))
+                  .Remote(1, 2);
+
+  EXPECT_EQ(obj5.Get(), 3);
+
+  auto obj6 = ray::Actor<Base>()
+                  .Remote()
+                  .Task(RayMemberFunc(&Base::overload_func, cv_none, int, int))
+                  .Remote(ObjectRef<int>{2}, 1);
+
+  EXPECT_EQ(obj6.Get(), 3);
+}
+
+TEST(RayApiTestV2, PolymorphicActorTask) {
+  auto base_obj = ray::Actor<Base>().Remote().Task(&Base::bar).Remote(1);
+  EXPECT_EQ(base_obj.Get(), 1);
+
+  auto derived_obj = ray::Actor<Derived>().Remote().Task(&Base::bar).Remote(1);
+  EXPECT_EQ(derived_obj.Get(), 2);
+}
+
+TEST(RayApiTestV2, TaskByName) {
+  int r = ray::Task("bar").Remote<int>(1).Get();
+  EXPECT_EQ(r, 1);
+
+  int r1 = ray::Actor<Base>().Remote().Task("&Base::bar").Remote<int>(2).Get();
```

Comment:
Why is there a need in name to mention `"&"`. Can it be just `Base::bar`

Reply:
There may be a problem, if we remove '&', maybe it represents a namespace, we can't distinguish whether it's a namespace or a class, the '&' indicates it's a class member function, not a namespace function.